Make eye(s) of god NODROP only when equipped

coiax opened the pull request:
:cl: coiax
tweak: The "eye of god" only becomes unremovable when put in your eye.
/:cl:

- The lavaland chest reward "eye of god", will only become NODROP (ie.
  unremovable) when actually placed in someone's glasses slot, rather
  than becoming immediately stuck in your hand.
- Eyes of god now hurt when implanted, causing temporary blindness and
  screaming.
- A pre-doubled eye of god is now a typepath.
- Eyes of god are no longer made out of glass.
- Glasses colouring procs have been moved to the top of the glasses
  file, but unchanged.

---
## Why It's Good For The Game

Thematically, it makes more sense that the eyes are unremovable only when implanted or "worn", and not that they're just made of magical superglue.
They're made of glass only because the basic glasses type are made of glass, but they're... fleshy eyes cut from a monster.
More screaming and messages gives the item a tiny bit more thematic colour.
A "doubled" type makes it easier for mappers and admins, if they care about such things.

coiax committed:
👀 Make eye(s) of god NODROP only when equipped

:cl: coiax
tweak: The "eye of god" only becomes unremovable when put in your eye.
/:cl:

- The lavaland chest reward "eye of god", will only become NODROP (ie.
  unremovable) when actually placed in someone's glasses slot, rather
  than becoming immediately stuck in your hand.
- Eyes of god now hurt when implanted, causing temporary blindness and
  screaming.
- A pre-doubled eye of god is now a typepath.
- Eyes of god are no longer made out of glass.
- Glasses colouring procs have been moved to the top of the glasses
  file, but unchanged.

TemporalOroboros commented:
👀

Fikou commented:
i feel like this is even more shitcode than the old version

nullAdept commented:
Could you also apply this change to the cursed ice hiking boots?

coiax commented:
@Fikou  Most of the "fiddlyness" comes from getting the grammar of the messages correct for the "painful insertion". Apart from that, isn't the structure about the same?

coiax commented:
> Could you also apply this change to the cursed ice hiking boots?

#55654

Fikou commented on code/modules/clothing/glasses/_glasses.dm:
```diff
@@ -434,57 +467,60 @@
-				H.remove_client_colour(old_colour_type)
-			if(glass_colour_type)
-				H.update_glasses_color(src, 1)
+	var/verbs = "burrows"
+	if(src.double)
+		verbs = "burrow"
 
+	to_chat(victim, "<span class='userdanger'>You experience blinding pain, as [src] [verbs] into your skull.</span>")
```
```suggestion
	to_chat(victim, "<span class='userdanger'>You experience blinding pain, as [src] [double ? "burrow" : "burrows"] into your skull.</span>")
```

ATH1909 commented on code/modules/clothing/glasses/_glasses.dm:
```diff
@@ -434,57 +467,60 @@
-					H.update_glasses_color(src, 1)
-	else
-		return ..()
+/obj/item/clothing/glasses/godeye/proc/pain(mob/living/victim)
+	// If pain/pain immunity ever implemented, check for it here.
```
bruh

Fikou commented on code/modules/clothing/glasses/_glasses.dm:
```diff
@@ -434,57 +467,60 @@
+				if(C.glasses == src)
+					pain(C)
+
+			src.name = initial(T.name)
+			src.desc = initial(T.desc)
+			src.double = TRUE
```
```suggestion
			name = initial(T.name)
			desc = initial(T.desc)
			double = TRUE
```

Fikou commented on code/modules/clothing/glasses/_glasses.dm:
```diff
@@ -434,57 +467,60 @@
+		if(!double)
+			var/obj/item/clothing/glasses/godeye/double/T = /obj/item/clothing/glasses/godeye/double
+
+			src.icon_state = initial(T.icon_state)
+			src.inhand_icon_state = initial(T.inhand_icon_state)
```
```suggestion
			icon_state = initial(T.icon_state)
			inhand_icon_state = initial(T.inhand_icon_state)
```

Commit:
Apply suggestions from code review

- Refactor out explicit `src` with implicit instance access.
- Use ternary string expression for burrow grammar

Co-authored-by: Fikou <piotrbryla@onet.pl>

coiax committed:
Fix NODROP staying on beheading

Even though the item is NODROP, it can still be removed from someone,
such as when they're beheaded. We don't want it to remain sticky, so
remove the NODROP trait.
